Library/Homebrew/download_strategy.rb

```diff
@@ -46,41 +50,57 @@ def stage
   sig { returns(String) }
   attr_reader :url
 
-  # Location of the cached download.
-  #
-  # @api public
   sig { returns(Pathname) }
-  attr_reader :cached_location
+  attr_reader :cache
 
-  attr_reader :cache, :meta, :name, :version
+  sig { returns(T::Hash[Symbol, T.untyped]) }
+  attr_reader :meta
+
+  sig { returns(String) }
+  attr_reader :name
+
+  sig { returns(T.any(NilClass, String, Version)) }
+  attr_reader :version
 
   private :meta, :name, :version
 
+  sig { params(url: String, name: String, version: T.any(NilClass, String, Version), meta: T.untyped).void }
   def initialize(url, name, version, **meta)
+    @cached_location = T.let(nil, T.nilable(Pathname))
```

Comment:
FYI: This broke custom strategies that implement:

```rb
def cached_location
  return @cached_location if defined?(@cached_location)

  [...]
end

Reply:
@krzysiek1507 Those custom strategies will need rewritten; we cannot avoid this pattern while using Sorbet. Instead use `return @cached_location if @cached_location`